File: solutions/go/collatz-conjecture/1/collatz_conjecture.go
package collatzconjecture

import "fmt"

func CollatzConjecture(n int) (int, error) {
  if n <= 0 {
      return 0, fmt.Errorf("must be positive")
  }
	steps := 0
    for n != 1 {
        if n%2 == 0{
            n = n / 2
        } else {
            n = n * 3 + 1
        }
        steps ++
    }
    return steps, nil
}

File: solutions/c/collatz-conjecture/1/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start){
    if (start <= 0) return -1;
    int steps = 0;
    int collats = start;
    while (collats != 1){
        if (collats % 2 ==0){
            collats = collats / 2;
        } else {
             collats = (collats * 3) + 1;
        }
        steps += 1;
    }
    return steps; 
}